index.js

```diff
@@ -119,14 +111,46 @@
   //  $$type :: String
   var $$type = '@@type';
 
-  //  type :: Any -> String
+  //  RPARSE :: RegExp
+  var RPARSE = /^(?:([^/\n\r]+)\/)?(.+?)(?:@(\d+))?$/;
+
+  //       TypeIdentifier :: (a, b, c) -> TypeIdentifier a b c
```

Comment:
Would that implicitly limit the total number of versions allowed to no more than 9?